### DISCUSSION TITLE
SSR-friendly useMediaQuery?

### DISCUSSION THREAD

**Discussion by ghost** [chakra-ui/chakra-ui#3788] [Ideas]
Hi, recently came across an example of useMediaQuery (outside of chakra) that is resilient against SSR. Here's a link: https://github.com/vercel/next.js/discussions/14810

It seems that this component utilizes useEffect and event listeners to check whether or not we are 'browser-side', and flips a boolean when we finally are.

Contrast this with the Chakra implmentation; it seems we simply exit early once we detect we are server-side or in a non-supporting environment: https://github.com/chakra-ui/chakra-ui/blob/75edcf41e7ff4acc2569f2169949063c164d8f6e/packages/media-query/src/use-media-query.ts

Do folks find the implementation in the first example something worth taking direction from and including in the chakra implementation? Wanted to get feedback from folks, especially from those who are more familiar with reasoning behind current implementation. If this seems like something worth pursuing I'd be happy to push a PR. Thanks!

**Comment by cyberwombat** [comment]
Thanks for the link! yes I don't know why this is not implemented. Never been able to use media/breakpoint hooks from chakra.
